README.md

```diff
@@ -1,7 +1,34 @@
 ## What is the difference between push, pull, and fetch?
 
-- `git push` - sent changes from a local branch to a remote repo
-- `git fetch` - get changes from a remote repo into your tracking branch
-- `git pull` - will get changes from a remote branch into your tracking branch and merge them into a local branch
+- `git fetch` - downloads changes from your _remote_ branch, and adds them to your
+  _tracking_ branch
+- `git pull` - downloads changes from your _remote_ branch, adds them to your
+  _tracking_ branch, and then merges them into your _tracking_ branch
+- `git push` - uploads changes from your _local_ or _tracking_ branch to your
+  _remote_ branch
+
+Often `git push` and `git pull` are described as equivalent. This isn't entirely
+correct, since `git pull` does two things.
+
+The `git fetch` command takes your current branch, and checks to see if there is
```

Comment:
I feel like this paragraph needs more "why".  Maybe you could have a discussion about feature branches vs main, `git diff` etc. to help the user know why they want or need to fetch. Whether they are tracking main or a feature branch has big implications with later parts of the workflow.